### ISSUE TITLE
`Mint.HTTP1` leaks socket upon receive timeout

### ISSUE THREAD

**Issue by fercasgal** [elixir-mint/mint#415]
When there's a receive timeout, the `Mint.HTTP1` simply does:

```elixir
defp handle_error(conn, error) do
  conn = put_in(conn.state, :closed)
  {:error, conn, error, []}
end
```

But the socket is still open and the calling code has no chance of closing it, since:

```elixir
def close(%MODULE{state: :closed} = conn) do
  {:ok, conn}
end
```

So the calling code believes that it has to open a new connection, and the previous one leaks.

**Comment by whatyouhide** [comment]
@fercasgal I pushed out #416, which _should_ fix this. If we get into `handle_error/2`, it means that there was a transport error, and I think we assumed that that meant the socket was closed.

**Comment by fercasgal** [comment]
I tried your branch in our setup and it fixed the memory leak we were seeing.

**Comment by whatyouhide** [comment]
Fantastic! I've released v1.5.2.
